pkg/apis/policy/validation/validation.go

```diff
@@ -50,8 +50,20 @@ func ValidatePodDisruptionBudgetUpdate(pdb, oldPdb *policy.PodDisruptionBudget)
 func ValidatePodDisruptionBudgetSpec(spec policy.PodDisruptionBudgetSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	allErrs = append(allErrs, extensionsvalidation.ValidatePositiveIntOrPercent(spec.MinAvailable, fldPath.Child("minAvailable"))...)
-	allErrs = append(allErrs, extensionsvalidation.IsNotMoreThan100Percent(spec.MinAvailable, fldPath.Child("minAvailable"))...)
+	if spec.MinAvailable != nil && spec.MaxUnavailable != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec, "MinAvailable and MaxUnavailable cannot be both set"))
+	}
+
```

Comment:
> @deads2k is there something that we need to know regarding cross-object validation or that is not a thing in the api-machinery plans?

Validation should be statically determinant.  If your object can be invalid, you should indicate that as a status condition in a way that guides the user towards a solution.  When the situation is resolved, you should update the status.  This makes more sense as you start to consider the implications of validating once and then the other object changes, making this invalid again.  If you try to fanout the validation you end up with a web of dependencies which are very hard to track.

The only exception we have in the API (that I know of), is done inside of a REST handler where the RBAC types check for escalation in order to prevent it.  This was done to prevent privilege escalation attacks and I don't know of another comparable use-case.